syntax = "proto3";

package gymu;

message UserRequest {
    int32 user_id = 1;
    repeated int32 ids = 2;
}

message IdsPackageMessage {
    repeated int32 ids = 1;
}

service BasketAnalyzeService {
    rpc RecommendExercises(UserRequest) returns (IdsPackageMessage) {}
    rpc RecommendWorkouts(UserRequest) returns (IdsPackageMessage) {}
}

message UserProgressRequest {
    int32 user_id = 1;
    int32 workout_id = 2;
}

message ExerciseItem {
    int32 exercise_id = 1;
    int32 item_id = 2;
    int32 order_index = 3;
    map<string, int32> value = 4;
}

message ProgressRecommendationMessage {
    repeated ExerciseItem exercises = 1;
}

service ProgressAnalyzeService {
    rpc RecommendProgress(UserProgressRequest) returns (ProgressRecommendationMessage) {}
}
